barretenberg/cpp/src/barretenberg/commitment_schemes/small_subgroup_ipa/small_subgroup_ipa.test.cpp

```diff
@@ -0,0 +1,250 @@
+#include "barretenberg/commitment_schemes/small_subgroup_ipa/small_subgroup_ipa.hpp"
+#include "barretenberg/commitment_schemes/pcs_test_flavors.hpp"
+#include "barretenberg/commitment_schemes/shplonk/shplemini.hpp"
+
+#include <array>
+#include <vector>
+
+namespace bb {
+template <typename Flavor> class SmallSubgroupIPATest : public ::testing::Test {
+  public:
+    using Curve = typename Flavor::Curve;
+    using Transcript = typename Flavor::Transcript;
+    using FF = typename Curve::ScalarField;
+
+    static constexpr size_t log_circuit_size = 7;
+
+    FF evaluation_challenge;
+
+    void SetUp() override { evaluation_challenge = FF::random_element(); }
+
+    /**
+     * @brief Comput the sum of the Libra constant term and Libra univariates evaluated at Sumcheck challenges.
+     *
+     * @param zk_sumcheck_data Contains Libra constant term and scaled Libra univariates
+     * @param multivariate_challenge Sumcheck challenge
+     * @param log_circuit_size
+     */
+    static FF compute_claimed_inner_product(ZKSumcheckData<Flavor>& zk_sumcheck_data,
+                                            const std::vector<FF>& multivariate_challenge,
+                                            const size_t& log_circuit_size)
+    {
+        const FF libra_challenge_inv = zk_sumcheck_data.libra_challenge.invert();
+        // Compute claimed inner product similarly to the SumcheckProver
+        FF claimed_inner_product = FF{ 0 };
+        size_t idx = 0;
+        for (const auto& univariate : zk_sumcheck_data.libra_univariates) {
```

Comment:
can't you iterate over the univariates and multivariate_challenge at the same time?